src/lib/fhirTables.js

```diff
@@ -397,3 +401,39 @@ function obsCompare(a, b) {
   return(effectiveB - effectiveA);
 }
 
+// +--------------------+
+// |    Plan of Care    |
+// +--------------------+
+
+
+function carePlanHeader() {
+  return (
+    <tr>
+      <th>Status</th>
+      <th>Intent</th>
+      <th>Activities</th>
+      <th>Category</th>
+      <th>Period Start</th>
+      {/* Add headers for other relevant CarePlan properties */}
+    </tr>
+  );
+}
+
+function carePlanRow(r, rmap, dcr) {
+  const status = r.status;
+  const intent = r.intent;
+  const activities = r.activity ? r.activity.map(activity => activity.detail.code.text).join(", ") : "";
```

Comment:
two thoughts here --- (1) either of activity.detail or activity.detail.code may be null, so need some checks here; (2) when rendering the code, use futil.renderCodeableJSX(activity, dcr) rather than just pulling from text (you'll have to update that join too since you'll be dealing with JSX elements instead of a string). By using this function we'll render optimally even if the user doesn't include a text element. For this to really make any difference we're going to have to include the full SNOMED IPS terminology in the app, but that's a separate issue (I'll add one).

Reply:
Thank you Sean! I added a joinJSXelement method to the fhirUtils file. Also added logic to check if the activity.detail or activity.detail.code are null. I did the same for category but not 100% sure if that was needed I can revert back if not.